components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -292,6 +292,27 @@ private void insertAccessToken(String accessToken, String consumerKey, AccessTok
                     throw new IdentityOAuth2Exception(errorMsg, e);
                 }
 
+                recoverFromConAppKeyConstraintViolation(accessToken, consumerKey, accessTokenDO,
+                        connection, userStoreDomain, retryAttemptCounter + 1);
+            } else {
+                throw new IdentityOAuth2Exception(
+                        "Error when storing the access token for consumer key : " + consumerKey, e);
+            }
+        } catch (Exception e) {
+            IdentityDatabaseUtil.rollbackTransaction(connection);
+            // Handle constrain violation issue in JDBC drivers which does not throw
+            // SQLIntegrityConstraintViolationException
```

Comment:
let's change the comment reflecting this Exception block block

Reply:
Fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1768/commits/0bd866a54ebe78ab50408a9f2203b449c56e6c79